packages/mui-base/src/Input/Input.tsx

```diff
@@ -147,8 +147,14 @@ const Input = React.forwardRef(function Input<RootComponentType extends React.El
   const InputComponent = multiline ? slots.textarea ?? 'textarea' : slots.input ?? 'input';
   const inputProps: WithOptionalOwnerState<InputInputSlotProps> = useSlotProps({
     elementType: InputComponent,
-    getSlotProps: (otherHandlers: EventHandlers) =>
-      getInputProps({ ...otherHandlers, ...propsToForward }),
+    getSlotProps: (otherHandlers: EventHandlers) => {
+      return getInputProps({
+        ...otherHandlers,
+        ...propsToForward,
```

Comment:
`otherHandlers` will have `eventHandlers` passed in `slotProps` where as `propsToForward` will have `eventHandlers` passed to `Input` component.

According to previous code, `eventHandlers` in `propsToForward` will always override `eventHandlers` passed in `slotProps`, this PR adds a small caveat to the functionality, where `eventHandlers` in `propsToForward` will only override if there is valid function available.